boa_engine/src/builtins/eval/mod.rs

```diff
@@ -81,14 +81,10 @@ impl Eval {
 
         // Because of implementation details the following code differs from the spec.
 
-        // Parse the script body (11.a - 11.d)
-        // TODO: Implement errors for 11.e - 11.h
-        let body = match context
-            .parse_eval(x.as_bytes(), direct, strict)
-            .map_err(|e| e.to_string())
-        {
+        // Parse the script body and handle early errors (6 - 11)
+        let body = match context.parse_eval(x.as_bytes(), direct, strict) {
             Ok(body) => body,
-            Err(e) => return context.throw_syntax_error(e),
+            Err(e) => return context.throw_syntax_error(e.to_string()),
```

Comment:
That works too, yes, but my thoughts were more on using the following:
```rust
map_err(ToString::to_string)
```

instead of adding a closure in `map_err`, but maybe it wasn't possible, due to by value/by reference function signatures.